src/main/java/ip/main/TaskList.java

```diff
@@ -0,0 +1,212 @@
+package ip.main;
+
+import ip.task.Deadline;
+import ip.task.Event;
+import ip.task.Task;
+import ip.task.Todo;
+
+import java.util.ArrayList;
+
+/**
+ * Contains the list of task
```

Comment:
Misleading comment. Your task list does a bunch of stuff in addition to maintaining the list of tasks.

In fact, this class is a bit too big.  You can consider segregating to multiple classes by identifying the different things you are doing here.